doc/source/developing/testing.rst

```diff
@@ -3,503 +3,397 @@
+
+If you are adding to a frontend that has tests already, simply add methods to the existing test class.
+
+There are several things that can make the test writing process easier:
+
+* ``yt/utilities/answer_testing/testing_utilities.py`` contains a large number of helper functions.
+* Most frontends end up needing to test much of the same functionality as other frontends. As such, a list of functions that perform such work can be found in ``yt/utilities/answer_testing/answer_tests.py``.
+* `Fixtures <https://docs.pytest.org/en/stable/fixture.html>`_! You can find the set of fixtures that have already been built for yt in ``$YT_GIT/conftest.py``. If you need/want to add additional fixtures, please add them there.
+* The `parametrize decorator <https://docs.pytest.org/en/stable/example/parametrize.html?highlight=parametrizing%20tests>`_ is extremely useful for performing iteration over various combinations of test parameters. It should be used whenever possible.
+    * The use of this decorator allows pytest to write the names and values of the test parameters to the generated answer files, which can make debugging failing tests easier, since one can easily see exactly which combination of parameters were used for a given test.
+    * It is also possible to employ the ``requires_ds`` decorator to ensure that a test does not run unless a specific dataset is found, but not necessary. If the dataset is parametrized over, then the ``ds`` fixture found in the root ``conftest.py`` file performs the same check and marks the test as failed if the dataset isn't found.
+
+Here is what a minimal example might look like for a new frontend:
 
 .. code-block:: python
 
-   from yt.utilities.answer_testing.framework import (
-       GenericImageTest,
-       requires_ds,
-       data_dir_load,
-   )
+    # Content of yt/frontends/new_frontend/tests/test_outputs.py
+    import pytest
 
-   from matplotlib import pyplot as plt
+    from yt.utilities.answer_testing.answer_tests import field_values
 
+    # Parameters to test with
+    ds1 = "my_first_dataset"
+    ds2 = "my_second_dataset"
+    field1 = ("Gas", "Density")
+    field2 = ("Gas", "Temperature")
+    obj1 = None
+    obj2 = ("sphere", ("c", (0.1, "unitary")))
 
-   @requires_ds(my_ds)
-   def test_my_ds():
-       ds = data_dir_load(my_ds)
+    @pytest.mark.answer_test
+    class TestNewFrontend:
+        answer_file = None
+        saved_hashes = None
+        answer_version = "000"
 
-       def create_image(filename_prefix):
-           plt.plot([1, 2], [1, 2])
-           plt.savefig("%s_lineplot" % filename_prefix)
+        @pytest.mark.usefixtures("hashing")
+        @pytest.mark.parametrize("ds", [ds1, ds2], indirect=True)
+        @pytest.mark.parametrize("field", [field1, field2], indirect=True)
+        @pytest.mark.parametrize("dobj", [obj1, obj2], indirect=True)
+        def test_fields(self, ds, field, dobj):
+            self.hashes.update("field_values" : field_values(ds, field, dobj))
```

Comment:
this should fix the error detected with pre-commit ci (blacken-docs can't parse this because it's not valid Python)
```suggestion
            self.hashes.update({"field_values" : field_values(ds, field, dobj)})
```